card/mod/02_basic_types/set/type/pointer.rb

```diff
@@ -4,6 +4,11 @@
   self.drop_item Env.params['drop_item'] if Env.params['drop_item']
 end
 
+event :insert_item_event, :before=>:approve, :on=>:save, :when=> proc {|c| Env.params['insert_item']} do
+  index = Env.params['item_index'] || 0
+  self.insert_item index.to_i, Env.params['insert_item']
+end
+
```

Comment:
Yes, assignment is the main place that it is ambiguous between creating a new local and sending a method.  Language-wise, it is a bit odd that an un-referenced local assignment has higher precedence than an assignment method, but it does.  It might be expensive to test for this case in all the places you would have to in order to get it right.

I can't recall any cases besides assignment and self.class to avoid the class keyword interpretation.